### PR TITLE
Ansible modules for PowerFlex version 2.3.0

### PR DESCRIPTION
**Pull request by kuttattz**
# Description
- Add support for PowerFlex ansible modules and roles on Azure.
- Add support for resource group provisioning to validate, deploy, edit, add nodes and delete a resource group.
- The Info module is enhanced to list out all the firmware repository.

# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, pep8, linting, or security issues
- [X] I have performed Ansible Sanity test using --docker default
- [X] I have verified that new and existing unit tests pass locally with my changes
- [X] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [X] Unit testing
- [X] Functional testing
- [X] Regression testing

### CODE REVIEW

**Review by meenakshidembi691 — APPROVED** [review]

File: docs/Release Notes.md [hunk 1]
```diff
@@ -28,25 +28,25 @@ Table 1. Revision history
...
 Along with the previous release deliverables, this release supports following features - 
-- Fault set module is introduced to create, get details, rename and delete fault sets.
-- The SDS module has been enhanced to facilitate SDS creation within a fault set.
-- The Info module is enhanced to retrieve lists related to fault sets, service templates, deployments, and managed devices.
+- Add support for resource group provisioning to validate, deploy, edit, add nodes and delete a resource group.
+- The Info module is enhanced to list out all the firmware repository.
+- Add support for PowerFlex ansible modules and roles on Azure.
 
 Known issues
 ------------
```
Comment (felixs88) [inline docs/Release Notes.md:52]: `Pagination in info module with offset and limit fetches more than expected records when listing service templates or deployments.`
to
Pagination in info module with offset and limit fetches more than expected records when listing service templates, deployments or firmware repository.

**Comment by ansible-collections-svc** [comment]
Can one of the admins verify this patch?

File: CHANGELOG.rst [hunk 2]
```diff
@@ -5,6 +5,21 @@ Dellemc.PowerFlex Change Logs
 .. contents:: Topics
 
 
+v2.3.0
+======
+
+Minor Changes
+-------------
+
+- Add support for PowerFlex ansible modules and roles on Azure.
```
Comment (trisha-dell) [inline CHANGELOG.rst:14]: Added support

File: changelogs/changelog.yaml [hunk 3]
```diff
@@ -146,3 +146,14 @@ releases:
         name: fault_set
         namespace: ''
     release_date: '2024-02-29'
+  2.3.0:
+    changes:
+      minor_changes:
+        - Add support for resource group provisioning to validate, deploy, edit, add nodes and delete a resource group.
```
Comment (trisha-dell) [inline changelogs/changelog.yaml:152]: Added fix wherever applicable.

Comment (kuttattz) [reply changelogs/changelog.yaml]: Done

File: CHANGELOG.rst (hunk 2 above)
Comment (kuttattz) [reply CHANGELOG.rst]: Done

File: docs/Release Notes.md (hunk 1 above)
Comment (kuttattz) [reply docs/Release Notes.md:52]: Done

**Review by trisha-dell — APPROVED** [review]

**Comment by felixs88** [comment]
image

**Review by felixs88 — APPROVED** [review]

File: playbooks/modules/info.yml [hunk 4]
```diff
@@ -88,7 +88,7 @@
         gather_subset:
           - firmware_repository
 
-    - name: Get the list of firmware repository with includes related, bundles, and components
+    - name: Get the list of firmware repository with includes
```
Comment (Jennifer-John) [inline playbooks/modules/info.yml:91]:
```suggestion
    - name: Get the list of firmware repository
```

File: playbooks/modules/info.yml [hunk 5]
```diff
@@ -100,7 +100,7 @@
         include_bundles: true
         include_components: true
 
-    - name: Get the list of firmware repository with filter, includes related, bundles and components
+    - name: Get the list of firmware repository with filter with includes
```
Comment (Jennifer-John) [inline playbooks/modules/info.yml:103]:
```suggestion
    - name: Get the list of firmware repository with filter
```

File: docs/modules/info.rst [hunk 6]
```diff
@@ -271,7 +271,7 @@ Examples
         gather_subset:
           - firmware_repository
 
-    - name: Get the list of firmware repository with includes related, bundles, and components
+    - name: Get the list of firmware repository with includes
```
Comment (Jennifer-John) [inline docs/modules/info.rst:274]:
```suggestion
    - name: Get the list of firmware repository
```

File: docs/modules/info.rst [hunk 7]
```diff
@@ -283,7 +283,7 @@ Examples
         include_bundles: true
         include_components: true
 
-    - name: Get the list of firmware repository with filter, includes related, bundles and components
+    - name: Get the list of firmware repository with includes
```
Comment (Jennifer-John) [inline docs/modules/info.rst:286]:
```suggestion
    - name: Get the list of firmware repository with filter
```

**Review by Jennifer-John — APPROVED** [review]

File: CHANGELOG.rst [hunk 8]
```diff
@@ -5,6 +5,21 @@ Dellemc.PowerFlex Change Logs
...
 
+v2.3.0
+======
+
+Minor Changes
+-------------
+
+- Added support for PowerFlex ansible modules and roles on Azure.
+- Added support for resource group provisioning to validate, deploy, edit, add nodes and delete a resource group.
+- The Info module is enhanced to list out all the firmware repository.
```
Comment (Jennifer-John) [inline CHANGELOG.rst:16]:
```suggestion
- The Info module is enhanced to list the firmware repositories.
```

File: changelogs/changelog.yaml [hunk 9]
```diff
@@ -146,3 +146,15 @@ releases:
         name: fault_set
         namespace: ''
     release_date: '2024-02-29'
+  2.3.0:
+    changes:
+      minor_changes:
+        - Added support for resource group provisioning to validate, deploy,
+          edit, add nodes and delete a resource group.
+        - The Info module is enhanced to list out all the firmware repository.
```
Comment (Jennifer-John) [inline changelogs/changelog.yaml:154]:
```suggestion
        - The Info module is enhanced to list the firmware repositories.
```

**Pull request merged**